### ISSUE TITLE
Uber's gomock 0.5.0 is incompatible with rules_go's gomock.bzl

### ISSUE THREAD

**Issue by TvdW** [bazel-contrib/rules_go#4153]
### What version of rules_go/gazelle/bazel/gomock are you using?

* rules_go 0.50.1 (latest)
* gazelle 0.39.1 (latest)
* bazel 7.3.1 (latest)
* Uber's [gomock 0.5.0](https://pkg.go.dev/go.uber.org/mock/gomock) (latest)

### What happened?

`reflect` mode got dropped, and so the latest release no longer generates a source file that then gets compiled into a binary, which rules_go expects it to do. Instead, a new `package` mode was added.

I tried to modify rules_go to support the new mode but my bazel-fu isn't quite at the level needed for that 😄

**Comment by linzhp** [comment]
We are not planning to support Uber's gomock in rules_go. The plan is to move it to the same repo as Uber's gomock so they can co-evolve easily. The maintainers of Uber's gomock agreed with the plan. It's in my backlog, but any helping hand will be welcomed.

**Comment by linzhp** [comment]
Yes, the first PR would be just that and maybe update the mockgen tool used by the rule to be Uber's mockgen. As followups, we should also
* set up some tests to make sure gomock rule work as expected
* publish the gomock rule to Bazel Central Registry

**Comment by jatin-scifin** [comment]
We are using Uber's gomock 0.5.0 in our project. Is there any timeline, when are we going to fix this?  We can move back to gomock 0.4.0, but given https://github.com/linzhp/uber-mock/blob/8f5f506c9a5405619843b28ec25137a7f037f7bf/bazel/rules/gomock.bzl#L22, I am not much in favor of using this gomock.bzl
```
# The rules in this files are still under development. Breaking changes are planned.
# DO NOT USE IT.
```

**Comment by jatin-scifin** [comment]
Is there any timeline for this? We are using uber gomock 0.5.0 in our codebase and this feature is really helpful.
Also, uber-go is not part of bazel central registry. Do I need to vendor to use gomock.bzl from v0.4.0?

**Comment by linzhp** [comment]
[bazel_gomock@0.2.0](https://registry.bazel.build/modules/bazel_gomock) is compatible with Uber's gomock v0.5.3-0.20250612195617-359202c7b2fe. Please migrate your gomock rule to that one. We are planning to remove the one in rules_go.

**Comment by mohitaron** [comment]
> [bazel_gomock@0.2.0](https://registry.bazel.build/modules/bazel_gomock) is compatible with Uber's gomock v0.5.3-0.20250612195617-359202c7b2fe. Please migrate your gomock rule to that one. We are planning to remove the one in rules_go.

Great I'm assuming you'll do the needful to add this to MODULE.bazel. Please let me know if you need help.

**Comment by linzhp** [comment]
Which MODULE.bazel?

**Comment by jatin-scifin** [comment]
We tried using [bazel_gomock@0.2.0](https://registry.bazel.build/modules/bazel_gomock). It worked well for normal interface, but not for generic interface. 
For ex. It will work good with 
```
type timepass interace {
}
```
Not good with 
```
type timepassGeneric[Res any] {
   xyz() (Res)
}
``` 
It won't generate anything for generic interaces.
